evalml/automl/automl_search.py

```diff
@@ -624,8 +652,50 @@ def __init__(
         self._best_pipeline = None
         self._searched = False
 
-        self.X_train = infer_feature_types(X_train)
-        self.y_train = infer_feature_types(y_train)
+        if self.holdout_set_size < 0 or holdout_set_size >= 1:
```

Comment:
I'm finding the flow of logic in these if statements to be confusing, would this following refactor cover all the bases?  Happy to run through this together to clarify these if blocks.
```
if self.holdout_set_size < 0 or holdout_set_size >= 1: # first validate if we're doing holdout

elif self.passed_holdout_set is False: # then check if holdout is not passed
     if len(X_train) >= self._HOLDOUT_SET_MIN_ROWS): # if > 500 create and log we're doing holdout
               ...
     else: # if 500 log and do nothing
             ....
else: # we're using passed in holdout set - log we're doing holdout.
```

Reply:
I think I adapted most of these changes. I left a separate log block below that prints out what the validation score is for clarity.